app/public/static/js/components/keypad.js

```diff
@@ -91,19 +92,32 @@ AFRAME.registerComponent("keypad", {
       this.solvedPuzzleEntity.setAttribute("visible", "true");
     } else {
       // Not solved yet
+      keypad_text = data.hasOwnProperty("keypad_text")
+        ? data.keypad_text
+        : "Enter Password";
+      keypad_text_color = data.hasOwnProperty("keypad_text_color")
+        ? data.keypad_text_color
+        : "white";
       el.setAttribute("super-keyboard", {
-        label: "Enter Password",
-        labelColor: "black",
+        label: keypad_text,
+        labelColor: keypad_text_color,
       });
     }
   },
 });
 
 async function removeError(el) {
+  const data = this.data.jsonData;
```

Comment:
Does `this` work here? Isn't `this` not defined? You may instead want to pass in the data in the method where we call `removeError`

Reply:
good catch